vite.config.js

```diff
@@ -1,45 +1,53 @@
-import { defineConfig } from 'vite';
+import {defineConfig, loadEnv} from 'vite';
 import react from '@vitejs/plugin-react';
 import nodePolyfills from 'rollup-plugin-node-polyfills';
 import NodeGlobalsPolyfillPlugin from '@esbuild-plugins/node-globals-polyfill';
 import NodeModulesPolyfillPlugin from '@esbuild-plugins/node-modules-polyfill';
 
 // https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [react()],
-  resolve: {
-    alias: {
-      // Polyfill Node.js core modules. Adjust the list as needed.
-      stream: 'stream-browserify',
-      url: 'url/',
-      assert: 'assert',
-      // Add other aliases if required by your dependencies
+export default defineConfig(({ mode }) => {
+  // https://vite.dev/config/#using-environment-variables-in-config
+  const env = loadEnv(mode, process.cwd());
+
+  return {
+    define: {
+      __APP_ENV__: JSON.stringify(env.APP_ENV),
+    },
```

Comment:
huh where is this from?

Reply:
Based on the official documentation:
- https://vite.dev/config/#using-environment-variables-in-config